### ISSUE TITLE
gdget -f json fails with TypeError

### ISSUE THREAD

**Issue by ikelly** [tienfuc/gdcmdtools#55]
```
$ gdget.py FILE_ID_REDACTED -f json
Traceback (most recent call last):
  File "/usr/local/bin/gdget.py", line 65, in <module>
    result = get.run()
  File "/usr/local/lib/python2.7/dist-packages/gdcmdtools/get.py", line 105, in run
    file_content = self.get_by_format(return_format[self.format])
TypeError: get_by_format() takes exactly 3 arguments (2 given)
```

**Comment by thinkle** [comment]
I'm seeing the same thing, fwiw.

**Comment by thinkle** [comment]
And here's a patch -- looked like a simple typo (that said, I'm now running into a separate issue, but I'll wait until I've dug further before posting on that):

diff -c /tmp/get.py /usr/local/lib/python2.7/dist-packages/gdcmdtools/get.py
*** /tmp/get.py	2015-10-14 21:44:17.623913045 -0400
--- /usr/local/lib/python2.7/dist-packages/gdcmdtools/get.py	2015-10-14 21:43:13.909943424 -0400
***************
*** 102,108 ****
                 self.save_as = title 
  
              if self.format == "json":
!                 file_content = self.get_by_format(return_format[self.format])
                  self.parse_gas_json(file_content, self.save_as)               
              else:
                  # FIXME: handle return value
--- 102,108 ----
                 self.save_as = title 
  
              if self.format == "json":
!                 file_content = self.get_by_format(self.save_as,return_format[self.format])
                  self.parse_gas_json(file_content, self.save_as)               
              else:
                  # FIXME: handle return value

Diff finished.  Wed Oct 14 21:45:36 2015

**Comment by thinkle** [comment]
And here's the complete fix:


diff -c /tmp/get.py /usr/local/lib/python2.7/dist-packages/gdcmdtools/get.py
*** /tmp/get.py	2015-10-14 21:44:17.623913045 -0400
--- /usr/local/lib/python2.7/dist-packages/gdcmdtools/get.py	2015-10-14 22:10:52.448583796 -0400
***************
*** 67,73 ****
                  file_fullname = "%s.%s" % (file_name, file_ext) 
  
                  with open(file_fullname, 'wb+') as f:
!                     f.write(file_source)
                  
                  j.pop("source")
                  new_json["files"].append(j)
--- 67,73 ----
                  file_fullname = "%s.%s" % (file_name, file_ext) 
  
                  with open(file_fullname, 'wb+') as f:
!                     f.write(file_source.encode('utf8')) # We need unicode!
                  
                  j.pop("source")
                  new_json["files"].append(j)
***************
*** 102,112 ****
                 self.save_as = title 
  
              if self.format == "json":
!                 file_content = self.get_by_format(return_format[self.format])
                  self.parse_gas_json(file_content, self.save_as)               
              else:
                  # FIXME: handle return value
!                 result, local_size = self.get_by_format(self.save_as, return_format[self.format])
                  if( result == False ):
                      raise Exception("File size check failed, download may be incompleted. local size is %d" % local_size)
  
--- 102,113 ----
                 self.save_as = title 
  
              if self.format == "json":
!                 result,file_content,local_size = self.get_by_format(self.save_as,return_format[self.format])
!                 print 'file_content=>',file_content
                  self.parse_gas_json(file_content, self.save_as)               
              else:
                  # FIXME: handle return value
!                 result,content,local_size = self.get_by_format(self.save_as, return_format[self.format])
                  if( result == False ):
                      raise Exception("File size check failed, download may be incompleted. local size is %d" % local_size)
  
***************
*** 188,197 ****
  
          if self.file_size:
              if( int(self.file_size) == local_size ):
!                 return True, local_size 
                  print "File size: %d" % local_size
              else:
!                 return False, local_size
          else:
              print "File size in bytes: %d" % local_size
!             return True, local_size
--- 189,198 ----
  
          if self.file_size:
              if( int(self.file_size) == local_size ):
!                 return True, response.content, local_size 
                  print "File size: %d" % local_size
              else:
!                 return False, response.content, local_size
          else:
              print "File size in bytes: %d" % local_size
!             return True, response.content, local_size

Diff finished.  Wed Oct 14 22:11:11 2015

**Comment by tienfuc** [comment]
It's fixed in abb1a5055bacd201a89aa3a35316c6b15b855f7b
